Extensions/Notification/notificationtools.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @memberof gdjs.evtTools
+ * @namespace
+ * @static
+ * @public
+ */
+
+//variables
+gdjs.evtTools.notification = {
+  status: 'default',
+  notif: [],        //used for actions
+  event: [],        //used for conditions (typically eventhandlers)
+};
+
+gdjs.evtTools.notification.requestPermission = function() {
+    Notification.requestPermission().then(function(status) {
+      gdjs.evtTools.notification.status = status;
+    });
+}
+
+gdjs.evtTools.notification.permissionGranted = function() {
+  if (Notification.permission === 'granted') {
+    return true
+  } else {
+    return false
+  }
+}
+
+gdjs.evtTools.notification.isSupported = function() {
+  //Check if the window object contains a Notification
+  if ('Notification' in window) return true
+}
+
+gdjs.evtTools.notification.createNotification = function(id, title, body, icon, lang, requireInteraction, silent, vibrate) {
+  //Notification configuration see https://developer.mozilla.org/en-US/docs/Web/API/Notification for more information
+  const config = {
+    body: body,
+    data: id,
+    icon: icon,
+    lang: lang,
+    requireInteraction: requireInteraction,
+    silent: silent,
+    vibrate: vibrate.split(",")
+  }
+
+  gdjs.evtTools.notification.notif[id] = new Notification(title, config);
+
+  gdjs.evtTools.notification.event[id] = gdjs.evtTools.notification.notif[id];
```

Comment:
What is this for? I think it would interfere with the rest and should be removed
```suggestion
```

Reply:
Originally I just had the notif array, but the notification object always got replaced with a boolean value. So if someone wanted check for a click, it would make an error message since the notification object didn't exist. To get around that I created a second array to handle the events and boolean values.